timeouts: Remove several in-tree usage of legacy timeouts

carlescufi opened the pull request:
Remove several instances of:

  select LEGACY_TIMEOUT_API

to make sure that we are not using it anywhere in the tree anymore.

carlescufi commented:
DNM yet, checking CI first.

nashif commented:
treewide is not avalid prefix for commits. Just put timeouts or something like that...

carlescufi commented:
> treewide is not avalid prefix for commits. Just put timeouts or something like that...

done

andyross reviewed (approved):
Approve in principle, but I'm pretty sure these aren't going to build absent a little API work to each subsystem.  The set of flags committed was (I'm pretty sure) the minimal set required to get CI to build.

In most cases the changes didn't look complicated, but would affect the external subsystem APIs.

carlescufi committed:
timeouts: Port BBC micro:bit to the new timeout API

Port the BBC micro:bit display driver to the new timeout API exposed by
the kernel.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
timeouts: Port UART async API to the new timeout API

Port the API to the new paradigm as to avoid having to select
LEGACY_TIMEOUT_API.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

nordic-krch commented:
changes look good but there are still CI failures showing other places where changes are needed.

carlescufi committed:
drivers: video: Remove legacy timeout use

Stop using the legacy timeout option in the video drivers.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
> changes look good but there are still CI failures showing other places where changes are needed.

Thanks @nordic-krch. I have reduced the scope of this PR and I am tracking instead the conversion here:
https://github.com/zephyrproject-rtos/zephyr/issues/24926

carlescufi committed:
soc: ti_simplelink: Port to the new timeout API

Port the usage of the timeouts to the new timeout API, in order to be
able to deselect the legacy timeout option.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@jhedberg can you review the micro:bit changes please?
@loicpoulain can you review the video changes please?
@vanti can you review the TI changes please?

jhedberg commented:
> @jhedberg can you review the micro:bit changes please?

They're fine. I'm not really happy about less intuitive looking API usage (in particular the loss of K_SECONDS) but I guess it's unavoidable.

carlescufi committed:
console: Port to the new timeout API

Port the code so that it uses the new timeout API.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
drivers: audio: Port to the new timeout API

Port the I2S and DMIC drivers to the new timeout API so that they do
not need to enable legacy timeouts.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
drivers: modem: Port to new timeout API

Port the internal modem API to the new timeout API, using the native
k_timeout_t.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

andyross commented:
Just refreshing my vote.  All the new code looks great.

vanti commented on soc/arm/ti_simplelink/Kconfig:
```diff
@@ -4,7 +4,6 @@
 
 config SOC_FAMILY_TISIMPLELINK
 	bool
-	select LEGACY_TIMEOUT_API
```
@carlescufi Some things will break as a result of this change, despite the changes in TI HAL.

One is the power management subsystem. I got a build error when building samples/boards/ti/cc13x2_cc26x2/system_off for cc1352r1_launchxl:

```
/home/vwan/zephyrproject-work/zephyr/subsys/power/policy/policy_residency_cc13x2_cc26x2.c:52:13: error: invalid operands to binary != (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
   52 |  if ((ticks != K_FOREVER) && (ticks < pm_min_residency[0])) {
      |             ^~
/home/vwan/zephyrproject-work/zephyr/subsys/power/policy/policy_residency_cc13x2_cc26x2.c:63:14: error: invalid operands to binary == (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
   63 |   if ((ticks == K_FOREVER) ||
      |              ^~
/home/vwan/zephyrproject-work/zephyr/subsys/power/policy/policy_residency_cc13x2_cc26x2.c:85:15: error: invalid operands to binary != (have 's32_t' {aka 'int'} and 'k_timeout_t' {aka 'struct <anonymous>'})
   85 |     if (ticks != K_FOREVER) {
      |               ^~
[39/140] Building C object zephyr/CMak...vers/pinmux/pinmux_cc13xx_cc26xx.c.obj
ninja: build stopped: subcommand failed.
```
I think it goes beyond the above however. In subsys/power/power.c for instance, I see 
`        _sys_suspend(K_FOREVER);`
Yet _sys_suspend() is supposed to be called with ticks. It would be good to review the PM code to see what else needs to change, as much of it is conditionally compiled.

Other errors show up with the wifi driver as well when I compile samples/net/sockets/http_get for cc3220sf_launchxl:

```
[1/170] Preparing syscall dependency handling

[42/170] Building C object zephyr/CMak...fi/simplelink/simplelink_sockets.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj 
ccache /home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.2.0-2277-g12cbbeb6bf40 -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../include -Izephyr/include/generated -I../../../../../soc/arm/ti_simplelink/cc32xx -I../../../../../lib/libc/newlib/include -I../../../../../subsys/net/lib/sockets/. -I../../../../../drivers/wifi/simplelink/. -I../../../../../subsys/net/lib/tls_credentials -I/home/vwan/zephyrproject-work/modules/hal/cmsis/CMSIS/Core/Include -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -Og -imacros/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/vwan/zephyrproject-work/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/vwan/zephyrproject-work=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj -MF zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj   -c /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c: In function 'simplelink_poll':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:544:12: error: invalid operands to binary == (have 'int' and 'k_timeout_t' {aka 'struct <anonymous>'})
  544 |  if (msecs == K_FOREVER) {
      |            ^~
[45/170] Building C object zephyr/CMak...ivers/wifi/simplelink/simplelink.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj 
ccache /home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.2.0-2277-g12cbbeb6bf40 -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../include -Izephyr/include/generated -I../../../../../soc/arm/ti_simplelink/cc32xx -I../../../../../lib/libc/newlib/include -I../../../../../subsys/net/lib/sockets/. -I../../../../../drivers/wifi/simplelink/. -I../../../../../subsys/net/lib/tls_credentials -I/home/vwan/zephyrproject-work/modules/hal/cmsis/CMSIS/Core/Include -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -Og -imacros/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/vwan/zephyrproject-work/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/vwan/zephyrproject-work=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj -MF zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink.c.obj   -c /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c: In function 'simplelink_scan_work_handler':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:114:48: error: incompatible type for argument 2 of 'k_delayed_work_submit'
  114 |   k_delayed_work_submit(&simplelink_data.work, delay);
      |                                                ^~~~~
      |                                                |
      |                                                s32_t {aka int}
In file included from ../../../../../include/zephyr.h:18,
                 from /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:10:
../../../../../include/kernel.h:3358:18: note: expected 'k_timeout_t' {aka 'struct <anonymous>'} but argument is of type 's32_t' {aka 'int'}
 3358 |      k_timeout_t delay)
      |      ~~~~~~~~~~~~^~~~~
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c: In function 'simplelink_mgmt_scan':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:148:48: error: incompatible type for argument 2 of 'k_delayed_work_submit'
  148 |   k_delayed_work_submit(&simplelink_data.work, delay);
      |                                                ^~~~~
      |                                                |
      |                                                s32_t {aka int}
In file included from ../../../../../include/zephyr.h:18,
                 from /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink.c:10:
../../../../../include/kernel.h:3358:18: note: expected 'k_timeout_t' {aka 'struct <anonymous>'} but argument is of type 's32_t' {aka 'int'}
 3358 |      k_timeout_t delay)
      |      ~~~~~~~~~~~~^~~~~
[46/170] Building C object zephyr/CMak...r/subsys/net/lib/sockets/sockets.c.obj
ninja: build stopped: subcommand failed.
```

carlescufi replied:
Thanks for the comment. I will take a look tomorrow and fix as many as I can find. If there are additional ones we can fix them during the stabilization period for 2.3.

jukkar commented on drivers/modem/modem_cmd_handler.h:
```diff
@@ -179,14 +180,14 @@ int modem_cmd_send(struct modem_iface *iface,
 				 struct modem_cmd_handler *handler,
 				 struct setup_cmd *cmds, size_t cmds_len,
-				 struct k_sem *sem, int timeout);
+				 struct k_sem *sem, int k_timeout_t);
```
Typo? `s/int k_timeout_t/k_timeout_t timeout/`

carlescufi replied:
oops, indeed. Thanks @jukkar

carlescufi commented:
@vanti I addressed your comments. There are two things that I'd like to mention:

>I think it goes beyond the above however. In subsys/power/power.c for instance, I see
_sys_suspend(K_FOREVER);

I agree, I didn't know about this one. I will take a look after this PR is merged.

>Other errors show up with the wifi driver as well when I compile samples/net/sockets/http_get for cc3220sf_launchxl:

Yep, I fixed this one.

pabigot commented on include/drivers/video.h:
```diff
@@ -326,15 +327,15 @@ static inline int video_enqueue(struct device *dev, enum video_endpoint_id ep,
 static inline int video_dequeue(struct device *dev, enum video_endpoint_id ep,
 				struct video_buffer **buf,
-				u32_t timeout)
+				k_timeout_t timeout)
```
Note that this is part of the video API which is marked as "stable".

carlescufi replied:
You are very right. I will undo this change I think, and just leave the current API as-is.

andyross replied:
I don't care much about the video API per se, but in fact the magic of CONFIG_LEGACY_TIMEOUT_API is that if you really need backwards source compatibility, you have it even with k_timeout_t.  So applications that need the stable API don't lose it, even if they need the new features.

Whether or not we care about video dequeue timers being high precision is sort of an open question though.  I'd expect they'd be happy with latency near ~30ms video frame time and wouldn't care.

carlescufi replied:
Actually, I will leave the change but add a note in the release notes and run it by everybody in the API meeting

carlescufi replied:
Yeah, honestly the main reason I switched to `k_timeout_t` is that this was `u32_t` in the first place, so it was wrong. I could've just as easily switched to `s32_t` for consistency with other driver APIs, perhaps it's just better if I do that?

pabigot commented on west.yml:
```diff
@@ -71,7 +71,7 @@ manifest:
       path: modules/hal/stm32
     - name: hal_ti
-      revision: 811363a384cbd2d2b699cae7b8c4375de8218b65
+      revision: ebf0a6f11cd6ff9f735db8391a6351ffeeda4cc3
```
Shouldn't this be in a separate commit?   Assuming there are HAL dependencies required for this (I don't see any...).

carlescufi replied:
Yes, this would be cleaner. I will fix this too.

carlescufi replied:
Actually, no. When I remove the legacy timeouts option from the `config SOC_FAMILY_TISIMPLELINK` every single sample will start to fail to build if the HAL itself is not updated. I will tweak the commits a bit, but this needs to be together at least with the removal of the Kconfig select.

pabigot replied:
I don't really see that; a west update commit could precede the removal of the default to use the old API, unless the HAL breaks with legacy timeout (which would not be a great solution path).  Not a blocker, the change is now explained.

carlescufi committed:
timeouts: Fix power _sys_suspend invocation

Use K_TICKS_FOREVER instead of K_FOREVER after the timeout API rework.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

pabigot commented on drivers/modem/modem_cmd_handler.c:
```diff
@@ -510,7 +511,8 @@ int modem_cmd_send_nolock(struct modem_iface *iface,
 			  struct modem_cmd *handler_cmds,
 			  size_t handler_cmds_len,
-			  const u8_t *buf, struct k_sem *sem, int timeout)
+			  const u8_t *buf, struct k_sem *sem,
```
It'd be nice to avoid formating changes in commits that make non-whitespace changes.

carlescufi replied:
> Change requested for the west manifest update in a commit that doesn't mention why it's changing. The rest is commentary.

Thanks for the review! I will ping you once I've pushed an update.

carlescufi commented:
@vanti 

>I think it goes beyond the above however. In subsys/power/power.c for instance, I see
_sys_suspend(K_FOREVER);

This one was never built, but I've fixed it now anyway.

pabigot reviewed (changes requested):
Change requested for the west manifest update in a commit that doesn't mention why it's changing.  The rest is commentary.

carlescufi commented on samples/bluetooth/mesh/src/microbit.c:
```diff
@@ -72,8 +73,7 @@ void board_prov_complete(void)
 
-	mb_display_image(disp, MB_DISPLAY_MODE_DEFAULT, K_SECONDS(10),
-			 &arrow, 1);
+	mb_display_image(disp, MB_DISPLAY_MODE_DEFAULT, 10 * 1000, &arrow, 1);
```
I'll do it manually, thanks

carlescufi committed:
drivers: wifi: Port the simplelink wifi to the new timeouts API

Port the code so it uses the new timeouts API.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
power: ti: Port to the new timeouts API

Port the TI-related code in subsys/power so it complies with the new
timeouts API.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

vanti commented:
> > Other errors show up with the wifi driver as well when I compile samples/net/sockets/http_get for cc3220sf_launchxl:
> 
> Yep, I fixed this one.

Thanks @carlescufi for the changes. There is one place left in the wifi driver when I build _samples/net/sockets/http_get_ for cc3220sf_launchxl:

```
[77/170] Building C object zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj 
ccache /home/vwan/bin/zephyr-sdk-0.11.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.2.0-2280-g54dbc10f2e59 -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../include -Izephyr/include/generated -I../../../../../soc/arm/ti_simplelink/cc32xx -I../../../../../lib/libc/newlib/include -I../../../../../subsys/net/lib/sockets/. -I../../../../../drivers/wifi/simplelink/. -I../../../../../subsys/net/lib/tls_credentials -I/home/vwan/zephyrproject-work/modules/hal/cmsis/CMSIS/Core/Include -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/. -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/home/vwan/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -Og -imacros/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/vwan/zephyrproject-work/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr/samples/net/sockets/http_get=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/vwan/zephyrproject-work/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/vwan/zephyrproject-work=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj -MF zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj   -c /home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c: In function 'simplelink_poll':
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:544:6: warning: implicit declaration of function 'MSECS' [-Wimplicit-function-declaration]
  544 |  if (MSECS(msecs) == K_FOREVER) {
      |      ^~~~~
/home/vwan/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:544:19: error: invalid operands to binary == (have 'int' and 'k_timeout_t' {aka 'struct <anonymous>'})
  544 |  if (MSECS(msecs) == K_FOREVER) {
      |      ~~~~~~~~~~~~ ^~
      |      |
      |      int
```

The rest of the errors seem posix related, which I think would be addressed in #24926.

carlescufi commented on drivers/wifi/simplelink/simplelink_sockets.c:
```diff
@@ -541,7 +541,7 @@ static int simplelink_poll(struct pollfd *fds, int nfds, int msecs)
 
 	/* Convert time to SlTimeval struct values: */
-	if (msecs == K_FOREVER) {
+	if (msecs == SYS_FOREVER_MS) {
```
@vanti and @pfalcon is this acceptable in a `_poll()` implementation, or should I hardcode `-1` instead if that is what POSIX says?

pfalcon replied:
I think it should be -1, yes. We should not mix domains of Zephyr kernel API and POSIX API, this way first can be adjusted without affecting use of 2nd.

pabigot replied:
From what I can tell this is from the fdtable abstraction, and supports "any I/O object implementing POSIX I/O semantics".  So -1 seems better.

BTW @carlescufi the analogous code in drivers/wifi/eswifi/eswifi_socket_offload.c and drivers/modem/modem_socket.c needs to be converted too, if that isn't being done somewhere else.  Both appear to pass an integer `msecs` to `k_sem_take`.

vanti replied:
Thanks @carlescufi for the looking into this. I agree with others that it should be -1. In fact, z_impl_zsock_poll() should probably have used -1 as well in case the value of SYS_FOREVER_MS changes, but that would belong in a different PR.

I now see that the remaining build errors are coming from posix.

carlescufi commented:
@pabigot could you revisit please?

pabigot commented on doc/releases/release-notes-2.3.rst:
```diff
@@ -98,6 +98,11 @@ Stable API changes in this release
+* Video
+
+  * The video_dequeue() API call now takes a k_timeout_t for the timeout
+    parameter. This defaults to s32_t if CONFIG_LEGACY_TIMEOUT_API is enabled.
```
It doesn "default to", it becomes (or "reverts to").

pabigot reviewed (approved):
OK.  The modem and eswifi fixes I mentioned in an earlier comment could be done separately.

carlescufi committed:
manifest: Update the TI HAL module

Update the TI HAL module so that it points to a revision that has been
ported to the new timeout API.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@pabigot thanks again. I have fixed the release notes and also split the west change into its own commit, since I did verify that indeed the changes were backwards compatible

vanti commented:
> @vanti I addressed your comments.

Looks good to me given the posix errors would be addressed separately. I have verified that I can now build samples/boards/ti/cc13x2_cc26x2/system_off for cc1352r1_launchxl. Only remaining item is the '-1'. I'll move out of the way in case you wish to merge.
